### ISSUE TITLE
Compatibility with websockets-pubsub

### ISSUE THREAD

**Issue by michielbdejong** [inrupt/solid-client-notifications-js#59]
Some Solid pod servers (like ESS) implement websockets-notification-servers.
Others (like NSS, PSS, CSS, ...) implement websockets-pubsub.

Will this library support both protocols or will it only specialise in websockets-notification-servers?

**Comment by nicolasmondada** [comment]
Hi Michiel,

Thanks for asking. Notification Streams in ESS is designed to support multiple protocols and webhooks is its first iteration with more to come in the future. We are making this available to everyone for free through PodSpaces and we are looking forward for people to try it out and provide their feedback.

Thanks,
Nick.-

**Comment by michielbdejong** [comment]
> websockets is its first iteration

You mean websockets-notification-services, not websockets-pubsub, right? So then the answer is 'no'?

**Comment by michielbdejong** [comment]
That means if I want to write a Solid app (instead of just an "Inrupt app", if that is a word) then I cannot use your library, because it will not work against NSS, PSS, CSS, etc. Have you considered that use case?

**Comment by nicolasmondada** [comment]
Yes, I'm sorry, I meant websockets, not webhooks. I corrected later.

There are only Solid apps :) and we know about this limitation, we are looking at different options for a convergence path that works for all parties involved.

**Comment by michielbdejong** [comment]
OK, next time maybe look at your options before releasing a product, but hopefully you can fix it in version 1.2!

For this version, if Inrupt client only works with Inrupt server, and not with other Solid servers, then maybe it would be fair to warn your users about this caveat. Something like #61 maybe?

**Comment by michielbdejong** [comment]
A possible way to fix it would be to create an equivalent client library for websockets-pubsub, and then a wrapper around the two, something like:
```
solid-client-notifications-js  solid-client-websockets-pubsub
                      |                       |
                      |                       |
                       \_____________________/
                                   |
                          solid-client-wrapper

**Comment by michielbdejong** [comment]
See https://github.com/solid/solid-spec/blob/master/api-websockets.md for more info on the websockets-pubsub protocol that NSS, PSS, and CSS implement. It may disappear from the scene one day, but for now, a Solid app will definitely need to support it if it wants to receive notifications from different Solid pod server implementations, other than your own one.

**Comment by nicolasmondada** [comment]
> OK, next time maybe look at your options before releasing a product, but hopefully you can fix it in version 1.2!

I think you misunderstand the situation here, we looked at the different options and decided to go with an implementation that provides with secure notifications, unlike [Solid Websockets API](https://github.com/solid/solid-spec/blob/master/api-websockets.md), which is implemented by NSS and others.

> For this version, if Inrupt client only works with Inrupt server, and not with other Solid servers, then maybe it would be fair to warn your users about this caveat. Something like #61 maybe?
Sure, I adjusted the text a bit to explain more about the decision behind the current incompatibility: https://github.com/inrupt/solid-client-notifications-js/pull/61#pullrequestreview-651320253

**Comment by scenaristeur** [comment]
is there a way for a potential library to know if it is acting on a ESS or on another version of Solid server ?
